packages/prisma-client/src/prisma.ts

```diff
@@ -16,13 +16,37 @@ declare global {
 
 const logPrisma = process.env.NODE_ENV === "production";
 
+const getPgBouncerUrl = () => {
+  // The URL constructor does not support the "postgresql" protocol, so we replace it with "https"
+  // to ensure that the connection is made over HTTPS instead of plain text.
+  const databaseUrl = new URL(
+    process.env.DATABASE_URL!.replace("postgresql://", "https://")
+  );
+
+  if (databaseUrl.hostname.endsWith("supabase.co")) {
```

Comment:
I don't like this hardcoding at all ...

Reply:
we shoudln't be bound to supabase in any way, in fact its likely that we are going to switch

Reply:
this should be completely coming from env

Reply:
I as devops don't like to have 100s of env variables.

Reply:
And here we don't bound to supabase, we just use pgbouncer if supabase database is used.